pkg/nfd-master/nfd-master.go

```diff
@@ -1161,7 +1203,7 @@ func createPatches(removeKeys []string, oldItems map[string]string, newItems map
 	// Determine items to add or replace
 	for key, newVal := range newItems {
 		if oldVal, ok := oldItems[key]; ok {
-			if newVal != oldVal {
+			if newVal != oldVal && overwrite {
```

Comment:
Heh, I was thinking some quite different meaning for the allowOverwrite setting –  but, what you're implementing here is actually pretty neat 😅

However, I don't think this now works quite as expected as values effectively become consts. NFD cannot change labels (or annotations or ERs) that it previously created. However, it definitely should be able to do so (if something in the system changes we want to reflect that in the labels).

So, I think the interpretation/logic should be so that `allowOverwrite=false` denies NFD from overwriting labels/annotations/ERs/taints that it does not own. Basically:

```golang
if oldVal, ok := oldItems[key]; ok {
    if newVal != oldVal && (!removeKeys.Has(key) || overwrite) {
        patches = append(patches, utils.NewJsonPatch("replace", jsonPath, key, newVal))
    }
}
```

Also, `removeKeys` is a REALLY bad name. Maybe change it to `ownedKeys` or something and change it to `sets.String`.

Thoughts?

Reply:
Yes, that makes sense. At first, I thought that NFD could change labels/annotations/ERs/taints that it previously created. Thanks for the information. I will update that.

Reply:
I would suggest changing the ownedKeys to a set of strings in another PR, to make this one more atomic. wdyt?